testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/AbstractClientPoliciesTest.java

```diff
@@ -445,21 +454,53 @@ protected String createClientByAdmin(String clientId, Consumer<ClientRepresentat
         return cId;
     }
 
-    protected ClientRepresentation getClientByAdmin(String cId) {
+    protected ClientRepresentation getClientByAdmin(String cId) throws ClientPolicyException {
         ClientResource clientResource = adminClient.realm(REALM_NAME).clients().get(cId);
-        return clientResource.toRepresentation();
+        try {
+            return clientResource.toRepresentation();
+        } catch (BadRequestException bre) {
+            processClientPolicyExceptionByAdmin(bre);
+        }
+        return null;
+    }
+
+    protected ClientRepresentation getClientByAdminWithName(String clientName) {
+        return adminClient.realm(REALM_NAME).clients().findByClientId(clientName).get(0);
     }
 
-    protected void updateClientByAdmin(String cId, Consumer<ClientRepresentation> op) {
+    protected void updateClientByAdmin(String cId, Consumer<ClientRepresentation> op) throws ClientPolicyException {
         ClientResource clientResource = adminClient.realm(REALM_NAME).clients().get(cId);
         ClientRepresentation clientRep = clientResource.toRepresentation();
         op.accept(clientRep);
-        clientResource.update(clientRep);
+        try {
+            clientResource.update(clientRep);
+        } catch (BadRequestException bre) {
+            processClientPolicyExceptionByAdmin(bre);
+        }
     }
 
-    protected void deleteClientByAdmin(String cId) {
+    protected void deleteClientByAdmin(String cId) throws ClientPolicyException {
         ClientResource clientResource = adminClient.realm(REALM_NAME).clients().get(cId);
-        clientResource.remove();
+        try {
+            clientResource.remove();
+        } catch (BadRequestException bre) {
+            processClientPolicyExceptionByAdmin(bre);
+        }
+    }
+
+    private void processClientPolicyExceptionByAdmin(BadRequestException bre) throws ClientPolicyException {
+        Response resp = bre.getResponse();
+        if (resp.getStatus() == Response.Status.BAD_REQUEST.getStatusCode()) {
```

Comment:
Can be simplified to 
```
Response resp = bre.getResponse();
if (resp.getStatus() != Response.Status.BAD_REQUEST.getStatusCode()) {
  resp.close();
  return;
}

String respBody = resp.readEntity(String.class);
Map<String, String> responseJson = null;
try {
    responseJson = JsonSerialization.readValue(respBody, Map.class);
} catch (IOException e) {
    fail();
}
throw new ClientPolicyException(responseJson.get(OAuth2Constants.ERROR), responseJson.get(OAuth2Constants.ERROR_DESCRIPTION));
```

Reply:
@valb3r I see, I'll fix as you suggested.